crates/dslab-storage/src/fs.rs

```diff
@@ -241,16 +241,47 @@ impl FileSystem {
     }
 
     /// Returns size of the file located at `file_path` if there is any.
-    pub fn get_file_size(&self, file_path: &str) -> Result<u64, String> {
+    pub fn file_size(&self, file_path: &str) -> Result<u64, String> {
         self.files
             .get(file_path)
             .ok_or(format!("file [{}] does not exist", file_path))
             .map(|f| f.size)
     }
 
-    /// Returns amount of used space on all disks currently used by this file system.
-    pub fn get_used_space(&self) -> u64 {
-        self.disks.iter().map(|(_, v)| v.borrow().get_used_space()).sum()
+    /// Returns amount of used space on all disks currently mounted to this file system.
+    pub fn used_space(&self) -> u64 {
+        self.disks.iter().map(|(_, v)| v.borrow().used_space()).sum()
+    }
+
+    /// Returns amount of free space on all disks currently mounted to this file system.
+    pub fn free_space(&self) -> u64 {
+        self.disks.iter().map(|(_, v)| v.borrow().free_space()).sum()
+    }
+
+    /// Returns cumulative capacity of all disks currently mounted to this file system.
+    pub fn capacity(&self) -> u64 {
+        self.disks.iter().map(|(_, v)| v.borrow().capacity()).sum()
+    }
+
+    /// Returns vec of disk info associated with mount points.
+    pub fn disks_info(&self) -> Vec<(String, DiskInfo)> {
+        self.disks
+            .iter()
+            .map(|(mount_point, disk)| (mount_point.to_owned(), disk.borrow().info()))
+            .collect()
+    }
+
+    /// Returns disk info for a mount point.
+    pub fn disk_info(&self, mount_point: &str) -> Result<DiskInfo, String> {
+        self.resolve_disk(mount_point).and_then(|disk| Ok(disk.borrow().info()))
```

Comment:
```suggestion
        self.resolve_disk(mount_point).map(|disk| disk.borrow().info())
```
См. https://rust-lang.github.io/rust-clippy/master/index.html#bind_instead_of_map

Reply:
Действительно, спасибо